### PR TITLE
Add Ring.app v0.55

### PR DESCRIPTION
**Pull request by anthonyangel**
#### Adding a new cask
- [x] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Commit message includes cask’s name.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

**Comment by reitermarkus** [comment]
As for the comment, `brew cask style` should display an error with a link to [doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment).

**Comment by anthonyangel** [comment]
I've amended this now and the checks pass, the code block on [https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md](url) has the comment`  # amazonaws.com is the official download host per the vendor homepage`, which is what I was originally using

**Comment by anthonyangel** [comment]
I've submitted a separate PR [https://github.com/caskroom/homebrew-cask/pull/23217](url) with an update to the docs

File: Casks/ring.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cask 'ring' do
+  version :latest
+  sha256 :no_check
+
+  # s3.amazonaws.com was verified as official when first introduced to the cask
```
Comment (reitermarkus) [inline Casks/ring.rb]: This is still too unspecific, take a look again at https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment.

**Comment by reitermarkus** [comment]
Please delete the parts of the pull request template that are not relevant and fill out the rest.

**Comment by adidalal** [comment]
You can use a versioned URL.

`version` should be `0.55`
`url` should be `"https://ring-mac-app-assets.s3.amazonaws.com/production/Ring_#{version}.zip"`
`appcast` is `https://ring-mac-app-assets.s3.amazonaws.com/production/ring-appcast.xml`

Needs a `# ring-mac-app-assets.s3.amazonaws.com was verified as official when first introduced to the cask` comment above the url ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

**Comment by anthonyangel** [comment]
Thanks for the help (I'm still new at this game), I've followed the advice, changed the comments, added a version, appcast and set the auto_updates flag. Hopefully it'll be 3rd time lucky.

File: Casks/ring.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  # ring-mac-app-assets.s3.amazonaws.com was verified as official when first introduced to the cask
+  url "https://ring-mac-app-assets.s3.amazonaws.com/production/Ring_#{version}.zip"
+  appcast 'https://ring-mac-app-assets.s3.amazonaws.com/production/ring-appcast.xml',
+          checkpoint: '56d4addc5e0302953537dfaff3cbd87283cc333e73d3b129bf251aa48a271f52'
+  name 'Ring'
+  homepage 'https://ring.com/'
+  license :unknown # TODO: change license and remove this comment; ':unknown' is a machine-generated placeholder
+
+  auto_updates 'True'
```
Comment (vitorgalvao) [inline Casks/ring.rb]: `auto_updates true` (all lowercase, no quoting).

**Comment by adidalal** [comment]
Thank you for the contribution. It needed some fixes, so they were made in e0be68faee64c4e4c29b45b9b9f9beb739f7d56a. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

**Pull request closed without merging**